src/rules/expect.ts

```diff
@@ -308,16 +342,20 @@ function parseAssertions(sourceFile: ts.SourceFile): Assertions {
     }
     // Match on the contents of that comment so we do nothing in a commented-out assertion,
     // i.e. `// foo; // $ExpectType number`
-    const match = /^ ?\$Expect(TypeSnapshot|Type|Error)( (.*))?$/.exec(commentMatch[1]) as
-      | [never, 'TypeSnapshot' | 'Type' | 'Error', never, string?]
+    const match = /^( *)(?:(\$Expect(?:TypeSnapshot|Type|Error))|(\^\?))(?: (.*))?$/.exec(commentMatch[1]) as
```

Comment:
I split the regex -- the original is unaltered, and the twoslash regex is much simpler!

The code for handling the twoslash comment is pretty long, so I factored it out into a separate function. I left the Expect code as-is since it's pretty stateful and factoring it out would be a bit messy.